Core/GDCore/Project/CustomBehavior.h

```diff
@@ -0,0 +1,40 @@
+#include "GDCore/Project/Behavior.h"
+#include "GDCore/Project/EventsBasedBehavior.h"
+#include "GDCore/Project/Project.h"
+#include "GDCore/Project/PropertyDescriptor.h"
+#include "GDCore/Serialization/Serializer.h"
+#include "GDCore/Serialization/SerializerElement.h"
+
+using namespace gd;
+
+/**
+ * \brief A gd::Behavior that stores its content in JSON and forward the
+ * properties related functions to Javascript with Emscripten.
+ */
+class CustomBehavior : public gd::Behavior {
+public:
+  CustomBehavior(const EventsBasedBehavior &eventsBasedBehavior_,
+                 const gd::String &fullType)
+      : // Name is not important as this object is just a "blueprint"
+        // that is copied (see calls to AddObject).
+        Behavior("CustomObject", fullType),
+        eventsBasedBehavior(eventsBasedBehavior_) {}
+  CustomBehavior *Clone() const override;
+
+  using Behavior::GetProperties;
+  using Behavior::InitializeContent;
+  using Behavior::UpdateProperty;
+
+protected:
+  virtual std::map<gd::String, gd::PropertyDescriptor>
+  GetProperties(const gd::SerializerElement &behaviorContent) const override;
+  virtual bool UpdateProperty(gd::SerializerElement &behaviorContent,
+                              const gd::String &name,
+                              const gd::String &value) override;
+  virtual void
+  InitializeContent(gd::SerializerElement &behaviorContent) override;
+
+private:
+  const EventsBasedBehavior &eventsBasedBehavior;
```

Comment:
This means we have to be careful not to have any CustomBehavior living in memory after their associated EventsBasedBehavior was deleted/removed.
Do you think of any situation where this could happen?

Reply:
It will be an issue when a behavior is removed. On the released version, removing an extension and adding it back preserve the behaviors on objects.
We could use a Project and a fullType instead of an EventsBasedBehavior. It will work the same way as before but hidden inside the CustomBehavior.

Reply:
I see! Might be safer. 
If we don't change anything (let the reference as you did now) and if we remove an extension, what would happen? Invalid reference => crash? These CustomBehavior are never recreated right, after being created once in objects?

Reply:
Yes, it doesn't do crash right away but it freezes some interactions with the UI and make stacks all over the place.